src/editor/components/modals/GeoModal/GeoModal.component.jsx

```diff
@@ -0,0 +1,151 @@
+      lng: roundToSix(event.latLng.lng())
+    }));
+  }, []);
+
+  const { isLoaded } = useJsApiLoader({
+    googleMapsApiKey: 'AIzaSyCwldpJKOZ1yh_FT8NrUPuPeMedEy1R2jw'
+  });
+
+  const onSaveHandler = () => {
+    if (!isProUser) {
+      onClose();
+    } else {
+      const latitude = markerPosition.lat;
+      const longitude = markerPosition.lng;
+      const elevation = markerPosition.elevation;
+      AFRAME.scenes[0].setAttribute('metadata', 'coord', {
+        latitude: latitude,
+        longitude: longitude,
+        elevation: elevation
+      });
+      const geoLayer = document.getElementById('reference-layers');
+      geoLayer.setAttribute(
+        'street-geo',
+        `latitude: ${latitude}; longitude: ${longitude}; elevation: ${elevation}`
+      );
+      // this line needs to update 3D tiles from the Editor. Need to delete after updating aframe-loaders-3dtiles-component
+      geoLayer.play();
+
+      onClose();
+    }
+  };
+
+  return (
+    <Modal
+      className={styles.modalWrapper}
+      isOpen={isOpen}
+      onClose={onClose}
+      extraCloseKeyCode={72}
+    >
+      <div className={styles.wrapper}>
+        <div className={styles.header}>
+          <img src={GeoImg} alt="geo" />
+          <p>Scene Location</p>
+          <p className={styles.badge}>Pro</p>
+        </div>
+        <Input
+          leadingIcon={<Mangnifier20Icon />}
+          placeholder="Search for a location"
+        ></Input>
+        {isLoaded && (
+          <GoogleMap
+            mapContainerStyle={{
+              width: '100%',
+              minHeight: '350px',
+              borderRadius: 4,
+              border: '1px solid #8965EF'
+            }}
+            center={{ lat: markerPosition.lat, lng: markerPosition.lng }}
+            zoom={1}
+            onClick={onMapClick}
+          >
+            <Marker
+              position={{ lat: markerPosition.lat, lng: markerPosition.lng }}
+            />
+          </GoogleMap>
+        )}
+
+        <div className={styles.sceneGeo}>
+          <div>
+            <p>Current Centerpoint</p>
+            <Input
+              leadingIcon={<p className={styles.iconGeo}>Lat, Long</p>}
+              tailingIcon={<Copy32Icon className={styles.copyIcon} />}
+              value={`${markerPosition.lat}, ${markerPosition.lng}`}
+              placeholder="None"
+            ></Input>
+          </div>
+          <div>
+            <p>New Centerpoint</p>
+            <Input
+              leadingIcon={<p className={styles.iconGeo}>Lat, Long</p>}
+              tailingIcon={<Copy32Icon className={styles.copyIcon} />}
+              placeholder="0, 0"
+            ></Input>
+          </div>
+        </div>
+
+        <div className={styles.controlButtons}>
+          <Button variant={'ghost'} onClick={onClose}>
+            Cancel
+          </Button>
+          <Button
+            trailingicon={
+              isProUser ? <></> : <span className={styles.locked}>🔒</span>
```

Comment:
Be aware that unicode emoticon may render differently on different OS and browser.